blog/_posts/2024-02-xx-scala-3.4.0-and-3.3.2-released.md

```diff
@@ -0,0 +1,187 @@
+- `-experimental` compiler flags will mark all top-level definitions as `@experimental`. This means that the experimental language features and definitions can be used in the project. Note that this does not change the strong guarantees of the stability of the non-experimental code. The experimental features can only be used in an experimental scope (transitively). In 3.5, we plan to allow using this flag also in the stable releases of the compiler.
+- If there is an error reading a class file, we now report the version of the classfile and a recommendation to check JDK compatibility:
+
+  ```
+  error while loading String,
+    class file /modules/java.base/java/lang/String.class is broken (version 65.0),
+    please check the JDK compatibility of your Scala version (3.4.0),
+    reading aborted with class java.lang.RuntimeException:
+    foo bar
+  ```
+
+  This will improve the user experience when the class file format changes unexpectedly. This feature will be backported to Scala 3.3.3.
+- Type inference for functions similar to fold is greatly improved. For example:
+
+  ```scala
+  def partition[T](xs: List[T], pred: T => Boolean) =
+     xs.foldRight((Nil, Nil)): (x, p) =>
+       if pred(x) then (x :: p._1, p._2) else (p._1, x :: p._2)
+
+  ```
+
+  will have its return type correctly inferred as `(List[T], List[T])`. Earlier, it would result in a rather unintuitive type error.
+- In 3.4, refutable patterns (i.e., patterns that might not match) in a for-comprehension generator must now be preceded by `case`, or an error is reported.
+
+  e.g.
+
+  ```scala
+  val xss = List(List(1, 2), List(3))
+
+  for y :: ys <- xss do println(y > 1)
+  //  ^^^^^^^
+  // error: pattern's type ::[Int] is more specialized than the right hand
+  // side expression's type List[Int]
+  //
+  // If the narrowing is intentional, this can be communicated by adding
+  // the `case` keyword before the full pattern, which will result in a filtering
+  // for expression (using `withFilter`).
+  ```
+
+  `.withFilter` is also no longer inserted for a pattern in a generator unless prefixed by `case`, meaning improved ergonomics for many types that do not implement `.withFilter`
+- Type inference has changed for `inline` methods in 3.4, which can cause a source incompatibility at the call site. The previous behaviour can be recovered in an affected file with
+
+  ```scala
+  import scala.language.`3.3`
+  ```
+
+  alternatively, the definition can changed to transparent inline, but as this is a TASTy breaking change, it is not a default recommendation. (Also, `inline` should be preferred when possible over `transparent inline` for reduced binary size)
+- JVM Backend parallelization has been ported from Scala 2.13 to Scala 3.
```

Comment:
A flag is needed to enable it, right? I'd suggest either saying what the flags is, and/or include a link to further information.